city_metrix/layers/building_classifier.py

```diff
@@ -0,0 +1,181 @@
+            EsaWorldCoverClass.CROPLAND.value: 1,
+            EsaWorldCoverClass.BUILT_UP.value: 2,
+            EsaWorldCoverClass.BARE_OR_SPARSE_VEGETATION.value: 3,
+            EsaWorldCoverClass.SNOW_AND_ICE.value: 4,
+            EsaWorldCoverClass.PERMANENT_WATER_BODIES.value: 4,
+            EsaWorldCoverClass.HERBACEOUS_WET_LAND.value: 4,
+            EsaWorldCoverClass.MANGROVES.value: 4,
+            EsaWorldCoverClass.MOSS_AND_LICHEN.value: 3
+            # Add other mappings as needed
+        }
+        reclassified_esa = xr.apply_ufunc(
+            np.vectorize(lambda x: reclass_map.get(x, x)),
+            esa_world_cover,
+            vectorize=True
+        )
+
+        reclassified_esa = reclassified_esa.rio.write_crs(esa_world_cover.rio.crs, inplace=True)
+
+        esa_1m = reclassified_esa.rio.reproject(
+            dst_crs=crs,
+            resolution=1,
+            resampling=Resampling.nearest
+        )
+
+        # Read ULU land cover, filter to city, select lulc band
+        ulu_lulc = UrbanLandUse(band='lulc').get_data(bbox)
+        ulu_roads = UrbanLandUse(band='road').get_data(bbox)
+        # Create road mask of 50
+        # Typical threshold for creating road mask 
+        road_mask = ulu_roads >= 50
+        ulu_lulc = ulu_lulc.where(~road_mask, 6)
+        # 1-Non-residential: 0 (open space), 1 (non-res)
+        # 2-Residential: 2 (Atomistic), 3 (Informal), 4 (Formal), 5 (Housing project)
+        # 3-Roads: 6 (Roads)
+        mapping = {0: 1, 1: 1, 2: 2, 3: 2, 4: 2, 5: 2, 6: 3}
+        for from_val, to_val in mapping.items():
+            ulu_lulc = ulu_lulc.where(ulu_lulc != from_val, to_val)
+        
+        # 1-Non-residential as default
+        ulu_lulc_1m = ulu_lulc.rio.reproject(
+            dst_crs=crs,
+            shape=esa_1m.shape,
+            resampling=Resampling.nearest,
+            nodata=1
+        )
+
+        # ANBH is the average height of the built surfaces, USE THIS
+        # AGBH is the amount of built cubic meters per surface unit in the cell
+        # https://ghsl.jrc.ec.europa.eu/ghs_buH2023.php
+        anbh = (ee.ImageCollection("projects/wri-datalab/GHSL/GHS-BUILT-H-ANBH_R2023A")
+                .filterBounds(ee.Geometry.BBox(*bbox))
+                .select('b1')
+                .mosaic()
+                )
+        ds = xr.open_dataset(
+            ee.ImageCollection(anbh),
+            engine='ee',
+            scale=100,
+            crs=crs,
+            geometry=ee.Geometry.Rectangle(*bbox)
+        )
+        anbh_data = ds.b1.compute()
+        # get in rioxarray format
+        anbh_data = anbh_data.squeeze("time").transpose("Y", "X").rename({'X': 'x', 'Y': 'y'})
+        
+        anbh_1m = anbh_data.rio.reproject(
+            dst_crs=crs,
+            shape=esa_1m.shape,
+            resampling=Resampling.nearest,
+            nodata=0
+        )
+
+        return buildings_sample, esa_1m, ulu_lulc_1m, anbh_1m, crs
```

Comment:
This function shouldn't return 4 arrays, it should return 1 array with the result of the building classifier (that's an array with classifications, right?). Otherwise, there's no way for people to actually use this class.